File: investigation/kotlin_probe/main_module/src/kz/greetgo/kotline_probe/main/launcher/BeanContainerMain.java
package kz.greetgo.kotline_probe.main.launcher;

import kz.greetgo.depinject.core.BeanContainer;
import kz.greetgo.depinject.core.Include;
import kz.greetgo.kotline_probe.main.beans.BeanConfigMain;
import kz.greetgo.kotline_probe.main.beans.MainBean;

@Include(BeanConfigMain.class)
public interface BeanContainerMain extends BeanContainer {
  MainBean main();
}

File: investigation/kotlin_probe/main_module/src/kz/greetgo/kotlin_probe/main/launcher/BeanContainerMain.kt
package kz.greetgo.kotlin_probe.main.launcher

import kz.greetgo.depinject.core.BeanContainer
import kz.greetgo.depinject.core.Include
import kz.greetgo.kotlin_probe.main.beans.BeanConfigMain
import kz.greetgo.kotlin_probe.main.beans.MainBean

@Include(BeanConfigMain::class)
interface BeanContainerMain : BeanContainer {
  fun main(): MainBean
}
